xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/pdf/impl/PdfExportImpl.java

```diff
@@ -322,31 +324,26 @@ String applyCSS(String html, String css, XWikiContext context)
     {
         LOGGER.debug("Applying the following CSS [{}] to HTML [{}]", css, html);
         try {
-            //System.setProperty("org.w3c.css.sac.parser", "org.apache.batik.css.parser.Parser");
-
             // Prepare the input
             Reader re = new StringReader(html);
             InputSource source = new InputSource(re);
             XHTMLDocumentFactory docFactory = XHTMLDocumentFactory.getInstance();
-            // Clear the default stylesheet loaded by CSS4J. CSS4J does this to simulate the browser (UA) stylesheet
-            // that is applied but in our case we don't display the resulting HTML in a browser but as input to
-            // transform it into PDF. We want to control all the CSS rules we apply and thus don't need nor want to
-            // have extra ones applied.
-            // See https://groups.google.com/forum/?#!topic/css4j/CVqzjz3ZXXQ for extra details.
-            docFactory.getStyleSheetFactory().getUserAgentStyleSheet(
-                CSSDocument.ComplianceMode.STRICT).getCssRules().clear();
             SAXReader reader = new SAXReader(docFactory);
             // Dom4J 2.1.1 disable external DTD by default which is required here, putting it back
             // See https://github.com/dom4j/dom4j/issues/51
-            reader.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", true);
+            try {
+                reader.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", true);
+            } catch (SAXNotSupportedException e) {
+            } catch (SAXNotRecognizedException e) {
```

Comment:
@carlosame could you explain why it's ok to ignore these 2 SAX errors? If external DTD loading is required by CSS4J then how would it work if there's a SAX error? Thx

Reply:
The catch clauses are introduced in case somebody specifies a non-Xerces parser in JAXP. The flows are the following:

A) JAXP gives a Xerces instance:

1. dom4j 2.1.1 sets `"http://apache.org/xml/features/nonvalidating/load-external-dtd"` to false, to disable DTD. No exception is thrown.
2. XWiki sets `"http://apache.org/xml/features/nonvalidating/load-external-dtd"` to true, to enable DTD loading. No exception.

B) JAXP gives a non-Xerces instance:
1. dom4j 2.1.1 sets `"http://apache.org/xml/features/nonvalidating/load-external-dtd"` to false. The feature is not recognized: exception is thrown and ignored.
2. XWiki sets `"http://apache.org/xml/features/nonvalidating/load-external-dtd"` to true. Exception is thrown (and was not being ignored). **Boom**.

Reply:
ok so the idea is that with a non-xerces parser, what dom4j does is just ignore in this case and thus we don't need to compensate for it. 

Got it! Thx, I'll add ample comments since it's not obvious at first sight :)